the-collab-lab/tcl-73-smart-shopping-list
# Create purchase urgency badges

## Description

Color coded urgency badges have been created to assist the user in understanding how soon they need to buy an item. The badges are displayed on the `List` page for each item.

## Related Issue

Closes #23 

## Acceptance Criteria

- [x] The `List` component renders color coded badges that displays the purchase urgency for an item
- [x] The badges are accessible


## Type of Changes
refactoring

## Updates

### Before
Text renders the purchase urgency
![Screenshot 2024-03-29 at 4 39 36 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/01fa55b9-cbd3-482b-ab69-d5ff9a90d9de)


### After
Text and color coded badge renders the purchase urgency
![Screenshot 2024-03-29 at 5 09 26 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/f26dfecf-5db0-4c2b-baf1-353fbd2bfc57)
![Screenshot 2024-03-29 at 5 10 50 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/07f6093b-58de-49cc-ac39-51714073d427)


## Testing Steps / QA Criteria
- git pull
- git checkout rs-style-purchase-priority-badges
- npm ci
- npm start
- Click on the `Coffee Supply` list name on the `Home` page
- Navigate to the `List` page
- Notice the color badges for overdue, soon, kind of soon, not so soon, and inactive
- Browse other lists and see these badges there as well. You may need to manipulate data in FireStore to see all of the badge colors.

Note: Styling of how the badge interacts with other items on this page will be handled when the `List` page is styled.
